packages/core/src/pubsub/outstanding-queries.ts

```diff
@@ -0,0 +1,129 @@
+import { StreamID } from '@ceramicnetwork/streamid'
+
+import {
+    PriorityQueue,
+    ICompare,
+} from '@datastructures-js/priority-queue';
+  
+
+/**
+ * Query Interface
+ */
+interface IQuery {
+    timestamp: number;
+    streamID: StreamID;
+    queryID: string;
+}
+
+/**
+ * Query abstraction with a timestamp, streamID, and queryID
+ */
+export class Query implements IQuery {
+    timestamp: number;
+    streamID: StreamID;
+    queryID: string;
+    constructor(t: number, sid: StreamID, qid: string){
```

Comment:
```suggestion
    constructor(readonly timestamp: number, readonly streamID: StreamID, readonly queryID: string){
```

Works as a shortcut for lines 22-24, 26-28. They are not needed anymore.

Reply:
@Jovonni can you please mark comments you addressed as resolved?